OptimizelySDK/Odp/OdpEventManager.cs

```diff
@@ -0,0 +1,406 @@
+            {
+                return;
+            }
+            
+            _logger.Log(LogLevel.DEBUG, "Stop requested.");
+
+            _flushIntervalCancellation.Cancel();
+
+            // one final time
+            FlushQueue();
+
+            State = ExecutionState.Stopped;
+
+            _logger.Log(LogLevel.DEBUG, $"Stopped. Queue Count: {_queue.Count}.");
+        }
+
+        /// <summary>
+        /// Associate a full-stack userid with an established VUID
+        /// </summary>
+        /// <param name="userId">Full-stack User ID</param>
+        public void IdentifyUser(string userId)
+        {
+            if (!IsOdpConfigurationReady())
+            {
+                return;
+            }
+            
+            var identifiers = new Dictionary<string, string>
+            {
+                {
+                    OdpUserKeyType.FS_USER_ID.ToString(), userId
+                },
+            };
+
+            var odpEvent = new OdpEvent(TYPE, "identified", identifiers);
+            SendEvent(odpEvent);
+        }
+
+        /// <summary>
+        /// Send an event to ODP
+        /// </summary>
+        /// <param name="odpEvent">ODP Event to forward</param>
+        public void SendEvent(OdpEvent odpEvent)
+        {
+            if (!IsOdpConfigurationReady())
+            {
+                return;
+            }
+            
+            if (State == ExecutionState.Stopped)
+            {
+                _logger.Log(LogLevel.WARN, "ODP is not enabled.");
+                return;
+            }
+
+            if (InvalidDataFound(odpEvent.Data))
+            {
+                _logger.Log(LogLevel.ERROR, "ODP data is not valid.");
+                return;
+            }
+
+            odpEvent.Data = AugmentCommonData(odpEvent.Data);
+            Enqueue(odpEvent);
+        }
+
+        /// <summary>
+        /// Add a new ODP event to the queue
+        /// </summary>
+        /// <param name="odpEvent"></param>
+        private void Enqueue(OdpEvent odpEvent)
+        {
+            if (_queue.Count >= _queueSize)
+            {
+                _logger.Log(LogLevel.WARN, $"ODP event send failed (queueSize = {_queue.Count}).");
+                return;
+            }
+
+            _queue.Enqueue(odpEvent);
+
+            ProcessQueue();
+        }
+
+        /// <summary>
+        /// Process the queue
+        /// </summary>
+        private void ProcessQueue()
+        {
+            if (State != ExecutionState.Running)
+            {
+                return;
+            }
+
+            _logger.Log(LogLevel.DEBUG, $"Processing Queue.");
+
+            State = ExecutionState.Processing;
+            while (QueueHasBatches())
+            {
+                DequeueSendSingleBatch();
+            }
+
+            State = ExecutionState.Running;
+        }
+
+        private void RegularlyFlushQueue()
+        {
+            while (!_flushIntervalCancellation.IsCancellationRequested)
```

Comment:
It's a good consideration.

Since `RegularlyFlushQueue` is wrapped in the Task `_flushQueueRegularly`, the `Start()` on the Task should prevent the private method from being executed multiple times.

Does that make sense?